### DISCUSSION TITLE
Does glob pattern support `{}`?

### DISCUSSION THREAD

**Discussion by jjangga0214** [vercel/turborepo#6976] [Help] [answered]
Hi!

For example, in many cases (e.g. eslint), 

```
**/*.{c,m,}{j,t}{x,}
```

means

`"c" or "m" or ""` * `"j" or "t"` * `"x" or ""`,

which is equal to 

```json
[
  "**/*.cjs",
  "**/*.mjs",
  "**/*.js",
  "**/*.cts",
  "**/*.mts",
  "**/*.ts",
  "**/*.cjsx",
  "**/*.mjsx",
  "**/*.jsx",
  "**/*.mtsx",
  "**/*.ctsx",
  "**/*.tsx"
]
```

`{}` is useful, especially when many pipelines are in the config.

It can significantly reduce line numbers.

Other popular libraries support this with slightly different syntaxs.

For example, the lib used by [lint-staged](https://www.npmjs.com/package/lint-staged) can be configured like this,

```
*.?({c,m}){j,t}s?(x)
```

where `?( )` means being optional, and `{  }` means OR.

Does turborepo support the feature?

Thanks.

**Comment by mehulkar** [comment, marked as answer]
I believe it does in some cases, we're using it [here](https://github.com/vercel/turbo/blob/61214a7309a9c5eeae67d843173b616af518998b/cli/turbo.json#L19) as well. We do know that this is an under-specified/documented feature, so please open an Issue for the cases where it doesn't work or behaves unexpectedly.
